packages/bcer-data-portal/app/src/hooks/useLeaflet.tsx

```diff
@@ -0,0 +1,338 @@
+} from '@/constants/localInterfaces';
+import useLocation from './useLocation';
+import { AppGlobalContext } from '@/contexts/AppGlobal';
+import { BcRouteLinkBuilder } from '@/util/bcRouteLink.util';
+import { GoogleMapLinkBuilder } from '@/util/googleMapLink.util';
+
+function useLeaflet(locationIds: string, config: LocationConfig) {
+  const [appGlobal, setAppGlobalContext] = useContext(AppGlobalContext);
+
+  const [routeData, setRouteData] = useState<BCDirectionData>();
+  const [directionError, setDirectionError] = useState<any>();
+
+  const {
+    selectedLocations,
+    setSelectedLocations,
+    removeSelectedLocationHandler,
+    addLocationToSelectedHandler,
+  } = useLocation(locationIds);
+
+  const [map, setMap] = useState<L.Map>();
+  const [markers, setMarkers] = useState<L.Marker[]>([]);
+  const [geoJsonData, setGeoJsonData] = useState<L.GeoJSON>();
+  const [startingLocation, setStartingLocation] =
+    useState<BCGeocoderAutocompleteData>();
+
+  // Initial routing options in the map controller
+  const initialRoutingOptions: RouteOptions = {
+    option: 'fastest',
+    roundTrip: false,
+    optimizeOrder: false,
+    ferrySchedule: false,
+    timeDependent: false,
+    traffic: false,
+    turnRestriction: false,
+    events: false,
+    crossingCost: false,
+    globalDistortionField: false,
+    turnCost: false,
+    localDistortionField: false,
+  };
+
+  const [routeOptions, setRouteOptions] = useState<RouteOptions>({
+    ...initialRoutingOptions,
+  });
+
+  /**
+   * Once Map is rendered in the dom this method must be called
+   * to set up map properties
+   */
+  const onRender = () => {
+    let m = L.map('map').setView([51.4, -116.9], 6);
+    // Saving map in state for future use
+    setMap(m);
+    addTileLayer(m);
+  };
+
+  /**
+   * Draws markers for all the locations
+   */
+  const drawAllMarkers = () => {
+    const mkrs: L.Marker[] = [];
+    mkrs.push(...drawStartingMarker());
+
+    if (selectedLocations?.length && map) {
+      selectedLocations.forEach((l) => {
+        if (l.latitude && l.longitude) {
+          let mkr = L.marker([+l.latitude, +l.longitude]);
+          mkr.bindTooltip(makeMarkerToolTip(l)).openTooltip();
+          mkrs.push(mkr);
+          mkr.addTo(map);
+        }
+      });
+    }
+    setMarkers(mkrs);
+    setMapBound(mkrs);
+  };
+
+  /**
+   * Appropriately zooms the map to show all markers
+   * @param mkrs
+   */
+  const setMapBound = (mkrs: L.Marker[]) => {
+    if (map && mkrs?.length) {
+      const group = new (L.featureGroup as any)(mkrs);
+      map.fitBounds(group.getBounds());
+    }
+  };
+
+  /**
+   * Draws a unique marker for the starting location
+   * @returns
+   */
+  const drawStartingMarker = (): L.Marker[] => {
+    if (startingLocation?.geometry?.coordinates && map) {
+      var startingIcon = new L.Icon({
+        iconUrl:
+          'https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-red.png',
```

Comment:
I think it would be good to bundle these with the app to make sure that they don't change (same with the shadow marker)